nano/node/active_transactions.cpp

```diff
@@ -579,6 +588,80 @@ void nano::active_transactions::erase (nano::block const & block_a)
 	}
 }
 
+bool nano::active_transactions::should_flush ()
+{
+	bool result (false);
+	counter.trend_sample ();
+	size_t minimum_size (1);
+	std::unique_lock<std::mutex> counter_lock (counter.mutex);
+	if (counter.rate == 0)
+	{
+		minimum_size = 4;
+	}
+	else
+	{
+		minimum_size = counter.rate * 10;
+	}
+	if (roots.size () > minimum_size)
+	{
+		if (counter.rate <= 10)
+		{
+			if (static_cast<double> (roots.size () * .75) < long_unconfirmed_size ())
```

Comment:
The static cats shouldn't be needed as the `.75` will make the resulting expression a double anyway.